optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/common/accessor/lambda/PropertySetterAccessor.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.domain.common.accessor.lambda;
+
+import java.lang.invoke.CallSite;
+import java.lang.invoke.LambdaConversionException;
+import java.lang.invoke.LambdaMetafactory;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.reflect.Method;
+
+public abstract class PropertySetterAccessor<T> {
+
+    private final Method setterMethod;
+    private final Class<?> propertyType;
+    private final MethodHandles.Lookup lookup;
+    private final T setterFunction;
+
+    public static PropertySetterAccessor create(final Method setterMethod,
```

Comment:
Just out of curiosity, @rsynek - do we have a reason to worry about `char[]`, `int[]` etc.? Would be interesting to see if this code works for them as well. (Although they should probably hide under the `Object` case.)

Reply:
yes, tried with int [] and it falls under Object